bolt/include/bolt/Core/BinaryContext.h

```diff
@@ -326,6 +326,9 @@ class BinaryContext {
   /// Returns true if DWARF4 or lower is used.
   bool isDWARFLegacyUsed() const { return ContainsDwarfLegacy; }
 
+  /// Returns true if DWARFUnit is valid.
+  bool isValidDwarfUnit(DWARFUnit *const DU) const;
```

Comment:
Since you are not handling `nullptr` as a special case, pass `DU` via reference (vs. a pointer).